core/src/libtx/proof.rs

```diff
@@ -228,9 +225,10 @@ where
 		let id = Identifier::from_serialized_path(depth, &msg[4..]);
 
 		let commit_exp = self.keychain.commit(amount, &id, &switch)?;
-		match commit == &commit_exp {
-			true => Ok(Some((id, switch))),
-			false => Ok(None),
+		if commit == &commit_exp {
+			Ok(Some((id, switch)))
+		} else {
+			Ok(None)
```

Comment:
This commit is also valid for the next occurrences of match replacement for if else.
This is valid but I'm wondering if it's more readable than the previous match case?

Reply:
I think clippy actually prefers you use if/else on a bool - 
https://users.rust-lang.org/t/is-it-bad-style-to-match-a-bool/14359